### PR TITLE
Document overrides with customSyntax

### LINKED ISSUE #5872: Document `overrides` and `customSyntax` working together
### What is the problem you're trying to solve?

My project has both less and styled-components. I want to use stylelint to lint both. For styled-components, I use `"customSyntax": "postcss-jsx"` in the .stylelintrc. For less, I use` "customSyntax": "postcss-less"`. This works well for the corresponding file type. But I can't have stylelint to lint both file types since I can only specify one value for `customSyntax`.

### What solution would you like to see?

perhaps allow specifying an array of `customSyntax` so that stylint can use different library to parse corresponding type?

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Closes #5872

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: docs/user-guide/configure.md [hunk 1]
```diff
@@ -252,6 +252,10 @@ In your `.stylelintrc.json`:
     "alpha-value-notation": "number"
   },
   "overrides": [
+    {
+      "files": ["*.scss", "**/*.scss"],
```
Comment (ybiquitous) [inline docs/user-guide/configure.md:256]: [ask] Are these patterns the same as below?

```suggestion
      "files": ["**/*.scss"],
```

**Review by ybiquitous — APPROVED** [review]
I've left a trivial comment, but LGTM 👍🏼

**Comment by ybiquitous** [comment]
Now I've retried the CI jobs.

File: docs/user-guide/configure.md (hunk 1 above)
Comment (jeddy3) [reply docs/user-guide/configure.md:256]: I don't think that pattern will match files in the root.

See: 
https://github.com/ota-meshi/stylelint-config-html/blob/af14c2bfa55eab72461f7047c25cc4f842b19996/html.js#L23

Comment (ybiquitous) [reply docs/user-guide/configure.md:256]: I got it. 👍🏼

**Pull request merged**